src/framework/preferences/js/Panels.js

```diff
@@ -528,28 +540,55 @@ var fluid_2_0 = fluid_2_0 || {};
         }) || value;
     };
 
-    fluid.prefs.compositePanel.rebaseTree = function (model, tree, memberName, modelRelayRules) {
-        var rebased = fluid.transform(tree, function (val, key) {
-            if (key === "children") {
-                return fluid.transform(val, function (v) {
-                    return fluid.prefs.compositePanel.rebaseTree(model, v, memberName, modelRelayRules);
-                });
-            } else if (key === "selection") {
-                return fluid.prefs.compositePanel.rebaseTree(model, val, memberName, modelRelayRules);
-            } else if (key === "ID") {
-                return fluid.prefs.compositePanel.rebaseID(val, memberName);
-            } else if (key === "parentRelativeID") {
-                return fluid.prefs.compositePanel.rebaseParentRelativeID(val, memberName);
-            } else if (key === "valuebinding") {
-                return fluid.prefs.compositePanel.rebaseValueBinding(val, modelRelayRules);
-            } else if (key === "value" && tree.valuebinding) {
-                var valuebinding = tree.valuebinding;
-                var modelValue = fluid.get(model, fluid.prefs.compositePanel.rebaseValueBinding(valuebinding, modelRelayRules));
-                return modelValue !== undefined ? modelValue : val;
-            } else {
-                return val;
-            }
-        });
+    fluid.prefs.compositePanel.rebaseTreeComp = function (msgResolver, model, treeComp, memberName, modelRelayRules) {
+        var rebased = fluid.copy(treeComp);
+
+        if (rebased.children) {
```

Comment:
I think this would look much clearer with "else" blocks expressing that only one branch is possible at a time - unless there is indeed some case whereby more than one could execute - which would be good to highlight if so!